File: livingdoc-fixture-converter/src/main/java/org/livingdoc/fixture/converter/common/BooleanConverter.java
package org.livingdoc.fixture.converter.common;

import java.lang.reflect.AnnotatedElement;

import org.livingdoc.fixture.api.converter.ConversionException;
import org.livingdoc.fixture.api.converter.TypeConverter;


public class BooleanConverter implements TypeConverter<Boolean> {

    @Override
    public Boolean convert(String value, AnnotatedElement element) throws ConversionException {
        String lowerCaseValue = value.trim().toLowerCase();
        if ("true".equals(lowerCaseValue)) {
            return true;
        }
        if ("false".equals(lowerCaseValue)) {
            return false;
        }
        throw new ConversionException("'" + value + "' cannot be converted to a Boolean");
    }

}

File: livingdoc-fixture-converter/src/main/java/org/livingdoc/fixture/converter/common/BooleanConverter.kt
package org.livingdoc.fixture.converter.common

import org.livingdoc.fixture.api.converter.ConversionException
import org.livingdoc.fixture.api.converter.TypeConverter
import java.lang.reflect.AnnotatedElement


open class BooleanConverter : TypeConverter<Boolean> {

    @Throws(ConversionException::class)
    override fun convert(value: String, element: AnnotatedElement?): Boolean {
        val trimmedLowerCaseValue = value.trim().toLowerCase()
        when (trimmedLowerCaseValue) {
            "true" -> return true
            "false" -> return false
        }
        throw ConversionException("Not a boolean value: '$value'")
    }

}
